google-cloud-firestore/src/test/java/com/google/cloud/firestore/OpenTelemetryOptionsTest.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.firestore;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.OpenTelemetry;
+import javax.annotation.Nullable;
+import org.junit.*;
+
+public class OpenTelemetryOptionsTest {
+  @Nullable private Firestore firestore;
+
+  @Before
+  public void setUp() {
+    GlobalOpenTelemetry.resetForTest();
+  }
+
+  @After
+  public void tearDown() {
+    if (firestore != null) {
+      firestore.shutdown();
+      firestore = null;
+    }
+  }
+
+  FirestoreOptions.Builder getBaseOptions() {
+    return FirestoreOptions.newBuilder().setProjectId("test-project").setDatabaseId("(default)");
+  }
+
+  @Test
+  public void defaultOptionsDisablesTelemetryCollection() {
+    FirestoreOptions firestoreOptions = getBaseOptions().build();
+    firestore = firestoreOptions.getService();
+    assertThat(firestore.getOptions().getOpenTelemetryOptions().getEnabled()).isFalse();
+    assertThat(firestore.getOptions().getOpenTelemetryOptions().getOpenTelemetry()).isNull();
+  }
+
+  @Test
+  public void canEnableTelemetryCollectionWithoutOpenTelemetryInstance() {
+    FirestoreOptions firestoreOptions =
+        getBaseOptions()
+            .setOpenTelemetryOptions(
+                FirestoreOpenTelemetryOptions.newBuilder().setTracingEnabled(true).build())
+            .build();
+    firestore = firestoreOptions.getService();
+    assertThat(firestore.getOptions().getOpenTelemetryOptions().getEnabled()).isTrue();
+    assertThat(firestore.getOptions().getOpenTelemetryOptions().getOpenTelemetry()).isNull();
```

Comment:
Will the GlobalOpenTelemetry be used in this case?

Reply:
Yes! I have added a check for that [in the next PR](https://github.com/googleapis/java-firestore/pull/1533/files#diff-6149176ee13427d4b65f991040da821a4f4a8fa0b7a4555b69a7106c73f7ddf2R72) (when it's actually possible to check it).